### ISSUE TITLE
Hi, things have been great if undisturbed

### ISSUE THREAD

**Issue by eueyrunbike** [svanhoutte/wyze_garmin_sync#22]
Hi, everything goes smooth when nothing is changed...but I recently moved. When I set up the same computer and started weighing in again, my connect would not receive the updated measurements. This is what I retrieved from the log. Hope you can help...again. Garmin has yet to release an index s3 and I refuse to buy the s2 after reading about all the problems. Your solution has gotten me through most of this time besides when there's a hiccup in my normal set up. I am also using a different router if that matters. Docker is on the latest version. My weight etc measurements were last updated on july 25... I couldn't get that far back on the logs. They seem to be repetitive with the errors so I just copied and pasted a portion. 
Log:

2024-09-05 18:01:36 wyzegarminconnect-1  | /wyze_garmin_sync /wyze_garmin_sync
2024-09-05 18:01:38 wyzegarminconnect-1  | Traceback (most recent call last):
2024-09-05 18:01:38 wyzegarminconnect-1  |   File "/wyze_garmin_sync/scale.py", line 19, in <module>
2024-09-05 18:01:38 wyzegarminconnect-1  |     response = Client().login(email=os.environ['WYZE_EMAIL'], password=os.environ['WYZE_PASSWORD'], key_id=os.environ['WYZE_KEY_ID'],api_key=os.environ['WYZE_API_KEY'])
2024-09-05 18:01:38 wyzegarminconnect-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-09-05 18:01:38 wyzegarminconnect-1  |   File "/usr/local/lib/python3.11/site-packages/wyze_sdk/api/client.py", line 184, in login
2024-09-05 18:01:38 wyzegarminconnect-1  |     response = self._auth_client().user_login(
2024-09-05 18:01:38 wyzegarminconnect-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-09-05 18:01:38 wyzegarminconnect-1  |   File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/auth_service.py", line 88, in user_login
2024-09-05 18:01:38 wyzegarminconnect-1  |     response = self.api_call(
2024-09-05 18:01:38 wyzegarminconnect-1  |                ^^^^^^^^^^^^^^
2024-09-05 18:01:38 wyzegarminconnect-1  |   File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/auth_service.py", line 58, in api_call
2024-09-05 18:01:38 wyzegarminconnect-1  |     return super().api_call(
2024-09-05 18:01:38 wyzegarminconnect-1  |            ^^^^^^^^^^^^^^^^^
2024-09-05 18:01:38 wyzegarminconnect-1  |   File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 347, in api_call
2024-09-05 18:01:38 wyzegarminconnect-1  |     return super().api_call(
2024-09-05 18:01:38 wyzegarminconnect-1  |            ^^^^^^^^^^^^^^^^^
2024-09-05 18:01:38 wyzegarminconnect-1  |   File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 204, in api_call
2024-09-05 18:01:38 wyzegarminconnect-1  |     return self.do_post(url=api_url, headers=headers, payload=json, params=params)
2024-09-05 18:01:38 wyzegarminconnect-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-09-05 18:01:38 wyzegarminconnect-1  |   File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 132, in do_post
2024-09-05 18:01:38 wyzegarminconnect-1  |     return self._do_request(client, req)
2024-09-05 18:01:38 wyzegarminconnect-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-09-05 18:01:38 wyzegarminconnect-1  |   File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 106, in _do_request
2024-09-05 18:01:38 wyzegarminconnect-1  |     raise err
2024-09-05 18:01:38 wyzegarminconnect-1  |   File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 93, in _do_request
2024-09-05 18:01:38 wyzegarminconnect-1  |     response.raise_for_status()
2024-09-05 18:01:38 wyzegarminconnect-1  |   File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
2024-09-05 18:01:38 wyzegarminconnect-1  |     raise HTTPError(http_error_msg, response=self)
2024-09-05 18:01:38 wyzegarminconnect-1  | requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://auth-prod.api.wyze.com/api/user/login
2024-09-05 18:01:38 wyzegarminconnect-1  | /wyze_garmin_sync/wyze_scale.fit: OK
2024-09-05 18:01:38 wyzegarminconnect-1  | no new measurment

**Comment by eueyrunbike** [comment]
Serves me right not to wait for your response. I noticed you had updated the docker-compose files along with the docker files. So I replaced my original files with those and entered all my credentials where needed. Then I did this and it didn't do much... sorry.

C:\Users\Monica\.docker>docker compose run --rm wyzegarminconnect
time="2024-09-05T18:49:29-07:00" level=warning msg="C:\\Users\\Monica\\.docker\\docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion"
/wyze_garmin_sync /wyze_garmin_sync
Traceback (most recent call last):
  File "/wyze_garmin_sync/scale.py", line 19, in <module>
    response = Client().login(email=os.environ['WYZE_EMAIL'], password=os.environ['WYZE_PASSWORD'], key_id=os.environ['WYZE_KEY_ID'],api_key=os.environ['WYZE_API_KEY'])
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/api/client.py", line 184, in login
    response = self._auth_client().user_login(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/auth_service.py", line 88, in user_login
    response = self.api_call(
               ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/auth_service.py", line 58, in api_call
    return super().api_call(
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 347, in api_call
    return super().api_call(
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 204, in api_call
    return self.do_post(url=api_url, headers=headers, payload=json, params=params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 132, in do_post
    return self._do_request(client, req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 106, in _do_request
    raise err
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 93, in _do_request
    response.raise_for_status()
  File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://auth-prod.api.wyze.com/api/user/login
md5sum: /wyze_garmin_sync/cksum.txt: No such file or directory
File ./wyze_scale.fit not found
file uploaded
md5sum: /wyze_garmin_sync/wyze_scale.fit: No such file or directory
2024-09-06 01:50:01: /wyze_garmin_sync ~
Traceback (most recent call last):
  File "/wyze_garmin_sync/scale.py", line 19, in <module>
    response = Client().login(email=os.environ['WYZE_EMAIL'], password=os.environ['WYZE_PASSWORD'], key_id=os.environ['WYZE_KEY_ID'],api_key=os.environ['WYZE_API_KEY'])
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/api/client.py", line 184, in login
    response = self._auth_client().user_login(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/auth_service.py", line 88, in user_login
    response = self.api_call(
               ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/auth_service.py", line 58, in api_call
    return super().api_call(
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 347, in api_call
    return super().api_call(
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 204, in api_call
    return self.do_post(url=api_url, headers=headers, payload=json, params=params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 132, in do_post
    return self._do_request(client, req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 106, in _do_request
    raise err
  File "/usr/local/lib/python3.11/site-packages/wyze_sdk/service/base.py", line 93, in _do_request
    response.raise_for_status()
  File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://auth-prod.api.wyze.com/api/user/login
md5sum: /wyze_garmin_sync/cksum.txt: no properly formatted MD5 checksum lines found
File ./wyze_scale.fit not found
file uploaded
md5sum: /wyze_garmin_sync/wyze_scale.fit: No such file or directory
docker compose run-d
docker compose logs -f

**Comment by svanhoutte** [comment]
Hi ,
I think you need to update the API key on your docker compose 
Did you create the api key here https://developer-api-console.wyze.com/#/apikey/view ?

**Comment by eueyrunbike** [comment]
I hasn’t changed the api key, but I’ll try that after I drop my kids off and report back. Thanks!Sent from my iPhoneOn Sep 6, 2024, at 7:17 AM, svanhoutte ***@***.***> wrote:﻿
Hi ,
I think you need to update the API key on your docker compose
Did you create the api key here https://developer-api-console.wyze.com/#/apikey/view ?

**Comment by eueyrunbike** [comment]
Ok I did got the new key id and api key. I edited the docker-compose file and open command prompt to enter:
docker compose run --rm wyzegarminconnect
docker compose up -d
docker compose logs -f
That seems to have solved it. I had two containers running. I deleted the top one. Thanks again for supporting this. Do I need to update the API key on a routine basis like annually or whenever I change my set up or router? Don't want to be a constant bother with my ignorance. If I'm suppose to update the key annually and perform what I just did, I can do that!
